### ISSUE TITLE
How to eliminate strange border effect in the CycleGAN output image

### ISSUE THREAD

**Issue by spinoza1791** [junyanz/pytorch-CycleGAN-and-pix2pix#1520]
Need tips/advice on how best to reduce the unwanted border effects from CycleGAN. See output image examples:
![epoch200_fake_B](https://user-images.githubusercontent.com/26606744/208502979-913eb6a0-0941-4a1f-aa4a-b909075c5bd5.png)
![epoch197_fake_B](https://user-images.githubusercontent.com/26606744/208503185-38a15312-c97c-4fa8-ae9e-657e692478e0.png)

Training command:
python train.py --dataroot /cyclegan-catdog_imp2_512-data/catdog_imp2_512 --name catdog_imp2_512 --model cycle_gan --netG resnet_9blocks --display_id -1 --batch_size 16 --gpu_ids 0,1,2,3 --load_size 512 --crop_size 512 --n_epochs 150 --n_epochs_decay 150 --init_type xavier --norm instance

**Comment by taesungp** [comment]
How many images did you use in the dataset? This dataset may be a challenging one because the animal faces are not aligned.

**Comment by junyanz** [comment]
Maybe you could use some data augmentation. Currently, both your load_size and crop_size are 512. Maybe use load_size 580 and crop_size 512. Certain artifacts will go away if the model is trained long enough.

**Comment by spinoza1791** [comment]
> Maybe you could use some data augmentation. Currently, both your load_size and crop_size are 512. Maybe use load_size 580 and crop_size 512. Certain artifacts will go away if the model is trained long enough.

Images are all 512x512, so would I still benefit from load_size 580? I trained out 400 epochs with same issue.

**Comment by spinoza1791** [comment]
> How many images did you use in the dataset? This dataset may be a challenging one because the animal faces are not aligned.

TrainA - 1300, TrainB - 500. Would you please elaborate on the alignment of the faces and how to improve the data?

**Comment by spinoza1791** [comment]
Found the solution here: https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/issues/782#issuecomment-538537477

Changing from instance to batch norm solves this specific border issue. The style transfer is less dramatic with batch norm apparently, but I will try to remedy this by tweaking other parameters.

**Comment by Wenting-Xu** [comment]
> Found the solution here: [#782 (comment)](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/issues/782#issuecomment-538537477)
> 
> Changing from instance to batch norm solves this specific border issue. The style transfer is less dramatic with batch norm apparently, but I will try to remedy this by tweaking other parameters.

 I had a similar problem. Have you found any other solutions?
